os/os_interactor.go

```
@@ -99,3 +98,32 @@ func (i DefaultInteractor) CopyFile(source, destination string) error {
 	}
 	return destFile.Close()
 }
+
+// IsWSL runs the "uname -a" command to see if the output contains "microsoft" and returns true if it does, else false.
+// This is so we can check if the Linux OS is actually Windows SubSystem for Linux, which requires
+// A different command to open a browser.
+func (i DefaultInteractor) IsWSL() bool {
+	b, err := exec.Command("uname", "-a").Output()
+	if err != nil {
+		fmt.Printf("\nWarning: %s\n", err.Error())
```

`stdout` gets caught by `kubectl`, so we should either return this error, or print it to `stderr`.

---

Yes that is true, I guess I was only printing a warning and going to let the code continue so that the `xdg-open` could attempt to run, which would either succeed or throw an error. I can change it to stop execution if you'd like.

---

It would probably be best to stop execution at this point.

---

code has been updated, thanks for the feedback!